aten/src/ATen/native/cuda/jit_utils.cpp

```diff
@@ -651,8 +690,9 @@ std::string generate_code(
   } else {
     env.s("bfloat16_string", "");
   }
-  if (dynamic_casting) {
-    env.s("complex_string", jiterator_complex_support_literal);
+  if (f_inputs_type == "std::complex<float>" || result_type == "std::complex<float>" ||
```

Comment:
This block is going to need a comment explaining what it's doing.

Reply:
Added